### PR TITLE
Move security manager override to its own module

### PR DESCRIPTION
**Pull request by vincbeck**
Move the security manager to its own module. In the near future, we'll need to add files specific to the security manager override and having a module for it makes it easier. There is no code change in this PR, files are only getting moved

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/auth/managers/fab/auth/__init__.py [hunk 1]
```diff
@@ -15,3 +14,252 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from __future__ import annotations
+
+import datetime
+
+# This product contains a modified portion of 'Flask App Builder' developed by Daniel Vaz Gaspar.
+# (https://github.com/dpgaspar/Flask-AppBuilder).
+# Copyright 2013, Daniel Vaz Gaspar
```
Comment (vandonr-amz) [inline airflow/auth/managers/fab/auth/__init__.py]: is that ok in open source code ?

I mean, there is an other license just above, I'm wondering if they wouldn't be in conflict ?

Comment (vincbeck) [reply airflow/auth/managers/fab/auth/__init__.py]: I am just moving code within Airflow codebase, so I guess there is no change here

Comment (vandonr-amz) [reply airflow/auth/managers/fab/auth/__init__.py]: oh ok didn't realize it was already there, I thought it was new code.

**Review by ferruzzi — APPROVED** [review]
LGTM

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**